### PR TITLE
Add show and hide password button to password field 

### PR DESCRIPTION
**Pull request by nicolaslee2775**
fix authgear/authgear-deploy#41

##### Creare password
image

##### Change password
image

### CODE REVIEW

File: resources/authgear/templates/__default__/web/change_password.html [hunk 1]
```diff
@@ -27,26 +27,42 @@ <h1 class="title primary-txt">{{ template "change-password-page-title" }}</h1>
...
+		data-password-policy-password=""
+		class="input text-input primary-txt"
+		type="password"
+		autocomplete="new-password"
+		autocapitalize="none"
+		name="x_new_password"
+		placeholder="{{ template "new-password-placeholder" }}"
+	>
+	<button class="show-password-button" type="button">
+		<i class="fas fa-eye" aria-hidden="true"></i>
```
Comment (louischan-oursky) [inline resources/authgear/templates/__default__/web/change_password.html:42]: We should add a localized `title` to the button.

```html
<button type="button class="show-password-button" title="{{ template "show-password-label" }}">
  <i class="fas fa-eye" aria-hidden="true" />
</button>
```

File: resources/authgear/static/authgear.js [hunk 2]
```diff
@@ -369,10 +369,44 @@ function setupPage() {
...
+
+    if (wrapper.classList.contains("show-password")) {
+      input.type = "text";
+    } else {
+      input.type = "password";
+    }
+
+    showPasswordButton.addEventListener("click", function(e) {
+      wrapper.classList.add("show-password");
+      input.type = "text";
```
Comment (louischan-oursky) [inline resources/authgear/static/authgear.js:388]: For completeness, add

```javascript
e.preventDefault();
e.stopPropagation();
```

at the beginning of the callback function body.

**Comment by nicolaslee2775** [comment]
@louischan-oursky Updated.

**Comment by louischan-oursky** [comment]
conflicted.

**Comment by nicolaslee2775** [comment]
@louischan-oursky Rebased.

**Pull request merged**